components/script/dom/gpudevice.rs

```diff
@@ -229,4 +236,181 @@ impl GPUDeviceMethods for GPUDevice {
             self.resolve_create_buffer_mapped(cx, buffer, array_buffer, wgpu_descriptor, valid)
         }
     }
+
+    /// https://gpuweb.github.io/gpuweb/#GPUDevice-createBindGroupLayout
+    fn CreateBindGroupLayout(
+        &self,
+        descriptor: &GPUBindGroupLayoutDescriptor,
+    ) -> DomRoot<GPUBindGroupLayout> {
+        #[derive(Clone)]
+        struct MaxLimits {
+            max_uniform_buffers_per_shader_stage: i32,
+            max_storage_buffers_per_shader_stage: i32,
+            max_sampled_textures_per_shader_stage: i32,
+            max_storage_textures_per_shader_stage: i32,
+            max_samplers_per_shader_stage: i32,
+        }
+        let mut storeBindings = HashSet::new();
+        // TODO: We should have these limits on device creation
+        let limits = GPULimits::empty();
+
+        let mut validation_map = HashMap::new();
+        let maxLimits = MaxLimits {
+            max_uniform_buffers_per_shader_stage: limits.maxUniformBuffersPerShaderStage as i32,
+            max_storage_buffers_per_shader_stage: limits.maxStorageBuffersPerShaderStage as i32,
+            max_sampled_textures_per_shader_stage: limits.maxSampledTexturesPerShaderStage as i32,
+            max_storage_textures_per_shader_stage: limits.maxStorageTexturesPerShaderStage as i32,
+            max_samplers_per_shader_stage: limits.maxSamplersPerShaderStage as i32,
+        };
+        validation_map.insert(
+            webgpu::wgpu::binding_model::ShaderStage::VERTEX,
+            maxLimits.clone(),
+        );
+        validation_map.insert(
+            webgpu::wgpu::binding_model::ShaderStage::FRAGMENT,
+            maxLimits.clone(),
+        );
+        validation_map.insert(
+            webgpu::wgpu::binding_model::ShaderStage::COMPUTE,
+            maxLimits.clone(),
+        );
+        let mut max_dynamic_uniform_buffers_per_pipeline_layout =
+            limits.maxDynamicUniformBuffersPerPipelineLayout as i32;
+        let mut max_dynamic_storage_buffers_per_pipeline_layout =
+            limits.maxDynamicStorageBuffersPerPipelineLayout as i32;
+        let mut valid = true;
+
+        let bindings = descriptor
+            .bindings
+            .iter()
+            .map(|bind| {
+                // TODO: binding must be >= 0
+                storeBindings.insert(bind.binding);
+                let visibility = match ShaderStage::from_bits(bind.visibility) {
+                    Some(visibility) => visibility,
+                    None => {
+                        valid &= false;
```

Comment:
This is the same as `valid = false`, right? If so, let's use that instead. Same comment about later uses in this method, too.

Reply:
You are right! Thank you!